token-metadata/js/test/state.test.ts

```diff
@@ -2,68 +2,31 @@ import { PublicKey } from '@solana/web3.js';
 import { expect } from 'chai';
 
 import type { TokenMetadata } from '../src/state';
-import { TOKEN_METADATA_DISCRIMINATOR, unpack } from '../src/state';
-import { getArrayEncoder, getBytesEncoder, getStructEncoder, getTupleEncoder } from '@solana/codecs-data-structures';
-import { getStringEncoder } from '@solana/codecs-strings';
+import { unpack, pack } from '../src';
 
 describe('Token Metadata State', () => {
-    const lengthBuffer = (buffer: Buffer | Uint8Array): Buffer => {
-        const length = Buffer.alloc(4);
-        length.writeUIntLE(buffer.length, 0, 4);
-        return length;
-    };
-
-    // Helper function to pack meta into tlv bytes slab
-    const pack = (meta: TokenMetadata) => {
-        const encoder = getStructEncoder([
-            ['updateAuthority', getBytesEncoder({ size: 32 })],
-            ['mint', getBytesEncoder({ size: 32 })],
-            ['name', getStringEncoder()],
-            ['symbol', getStringEncoder()],
-            ['uri', getStringEncoder()],
-            ['additionalMetadata', getArrayEncoder(getTupleEncoder([getStringEncoder(), getStringEncoder()]))],
-        ]);
-        const data = encoder.encode({
-            ...meta,
-            updateAuthority: meta.updateAuthority?.toBuffer(),
-            mint: meta.mint.toBuffer(),
-        });
-        return Buffer.concat([TOKEN_METADATA_DISCRIMINATOR, lengthBuffer(data), data]);
-    };
-
-    it('Can unpack', () => {
-        const data = Buffer.from([
-            // From rust implementation
-            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 0, 0, 0, 110, 97,
-            109, 101, 6, 0, 0, 0, 115, 121, 109, 98, 111, 108, 3, 0, 0, 0, 117, 114, 105, 0, 0, 0, 0,
-        ]);
-
-        const input = Buffer.concat([TOKEN_METADATA_DISCRIMINATOR, lengthBuffer(data), data]);
-
-        const meta = unpack(input);
-        expect(meta).to.deep.equal({
+    it('Can pack and unpack as rust implementation', () => {
+        const meta = {
```

Comment:
I just noticed this test doesn't use `updateAuthority` in the data structure. Can you add to the test a step where you add `updateAuthority` and test the bytes again? It doesn't have to be a separate test.

Reply:
I modified the last test which has the updateAuthority to check against the bytes. 

https://github.com/solana-labs/solana-program-library/pull/5806/commits/3839116439f00552319248f01d2af4206ae954ae#diff-3223c36003634dbc6a302f5e66b19f6aed12d8aa50040ebf6b79ec4fa3a45c5fR52